.github/workflows/build-and-tag-maven.yml

```diff
@@ -49,6 +49,7 @@ jobs:
           echo "VERSION=$VERSION" >> $GITHUB_ENV
 
       - name: Login to DockerHub Registry
+        if: ${{ github.event_name == 'push' }}
```

Comment:
I am quite surprised to see this inside a workflow that can be invoked only by `workflow_call`. Does really inherit the event of the caller?

Reply:
There are multiple existing checks like this one on this build, and yes, it seems to be inherited.

Reply:
One thing that could make this clearer is to pass the "should push" boolean as an input parameter. We did discuss it here: https://github.com/Alfresco/alfresco-build-tools/pull/71#discussion_r891976829 but this was postponed.
I can do the change now (as well as put all secrets as inputs, so that I don't update all the multiple callers of this reusable work multiple times).